### PR TITLE
stylelint: add types for creating custom rules

### PR DESCRIPTION
**Pull request by filips-alpe**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:

- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

**Comment by filipsalpe** [comment]
Covers https://github.com/stylelint/stylelint/blob/master/lib/index.js#L23-L28 and https://github.com/stylelint/stylelint/blob/master/lib/index.js#L33

File: types/stylelint/index.d.ts [hunk 1]
```diff
@@ -50,3 +52,35 @@ export namespace formatters {
...
+        ruleName: string;
+        result: any;
+        message: string;
+        node: any;
+        index?: number;
+        word?: string;
+        line?: number;
+    }): void;
+
+    function ruleMessages(ruleName: string, messages: { [key: string]: any; }): typeof messages;
```
Comment (filipsalpe) [inline types/stylelint/index.d.ts]: For TS<=2.1 `typeof messages` would have to be changed to something like `{ [key: string]: any; }` or `object`

**Review by alan-agius4 — CHANGES_REQUESTED** [review]
Left feedback

**Comment by filipsalpe** [comment]
@alan-agius4 Thanks, updated!

**Comment by filipsalpe** [comment]
@RyanCavanaugh For some reason @typescript-bot thinks that another revision is needed...

**Pull request merged**